### PR TITLE
Add Consul Enterprise section to Docs

### PR DESCRIPTION
**Pull request by rcgenova**
Adds a Consul Enterprise section to the Docs site that basically indexes the Enterprise features in one place. This makes Consul Enteprise more discoverable in general and also enables us to send a single link to folks who'd like to learn more about the enterprise product. This mirrors the Enterprise section of the Vault docs site:  https://www.vaultproject.io/docs/vault-enterprise/index.html

### CODE REVIEW

File: website/source/docs/consul-enterprise/backups/index.html.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+page_title: "Consul Enterprise Automated Backups"
+sidebar_current: "docs-consul-enterprise-backups"
+description: |-
+  Consul Enterprise provides a highly available service that manages taking snapshots, rotation and sending backup files offsite to Amazon S3.
+
+---
+
+# Consul Enterprise Automated Backups
+
+Consul's core snapshot functionality allows you to save and restore the state of the Consul servers for disaster recovery. Snapshots are atomic and point-in-time, and  include key/value entries, service catalog, prepared queries, sessions, and ACLs.
```
Comment (sethvargo) [inline website/source/docs/consul-enterprise/backups/index.html.md]: We should wrap at 80 characters - it makes reviewing easier

File: website/source/docs/consul-enterprise/backups/index.html.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (sethvargo) [inline website/source/docs/consul-enterprise/backups/index.html.md]: I think the docs should live at /enterprise, not /consul-enterprise. Otherwise the URL is https://www.consul.io/docs/consul-enterprise. I think /docs/enterprise is a bit cleaner. I know Vault uses vault-enterprise.

File: website/source/docs/consul-enterprise/backups/index.html.md (hunk 1 above)
Comment (sethvargo) [inline website/source/docs/consul-enterprise/backups/index.html.md]: (There are editors for most major plugins to help)

File: website/source/docs/consul-enterprise/index.html.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+description: |-
+  Consul Enterprise features a number of capabilities beyond the open
+  source offering that may be beneficial in certain workflows.
+---
+
+# Consul Enterprise
+
+Consul Enterprise simplifies operations by automating workflows. It adds support for microservices deployments across complex network topologies. It also increases both scalability and resilience. Features include:
+
+- [Automated Backups](/docs/consul-enterprise/backups)
```
Comment (sethvargo) [inline website/source/docs/consul-enterprise/index.html.md]: These should end with /index.html (it'll work without, but we get penalized for duplicate content if we don't link directly to the page)

File: website/source/docs/consul-enterprise/upgrades/index.html.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+layout: "docs"
+page_title: "Consul Enterprise Automated Upgrades"
+sidebar_current: "docs-consul-enterprise-upgrades"
+description: |-
+  Consul Enterprise supports an upgrade pattern that allows you to deploy a complete cluster of new servers and then just wait for the upgrade to complete.
+---
+
+# Consul Enterprise Automated Upgrades
+
+[Consul Enterprise](https://www.hashicorp.com/consul.html) supports an [upgrade pattern](https://www.consul.io/docs/guides/autopilot.html#upgrade-migrations) that allows you to deploy a complete cluster of new servers and then just wait for the upgrade to complete. As the new servers join the cluster, server introduction logic checks the version of each Consul server. If the version is higher than the version on the current set of voters, it will avoid promoting the new servers to voters until the number of new servers matches the number of existing servers at the previous version. Once the numbers match, Autopilot will begin to promote new servers and demote old ones.
```
Comment (sethvargo) [inline website/source/docs/consul-enterprise/upgrades/index.html.md]: Try to avoid first person ("you" in this case)

File: website/source/layouts/docs.erb [hunk 5]
```diff
@@ -246,6 +246,30 @@
       <li<%= sidebar_current("docs-faq") %>>
         <a href="/docs/faq.html">FAQ</a>
       </li>
+
+      <hr>
+
+      <li<%= sidebar_current("docs-consul-enterprise") %>>
```
Comment (sethvargo) [inline website/source/layouts/docs.erb]: Same thing as I said above, I'd shorten this to docs-enterprise and update everywhere

**Comment by rcgenova** [comment]
@sethvargo all comments addressed in https://github.com/hashicorp/consul/pull/3008/commits/f20aad898349a379d38a48487e9aaa747254815b

File: website/source/docs/enterprise/backups/index.html.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+sidebar_current: "docs-enterprise-backups"
+description: |-
+  Consul Enterprise provides a highly available service that manages taking snapshots, rotation and sending backup files offsite to Amazon S3.
+---
+
+# Consul Enterprise Automated Backups
+
+Consul's core snapshot functionality allows operators to save and restore the state of
+the Consul servers for disaster recovery. Snapshots are atomic and
+point-in-time, and  include key/value entries, service catalog, prepared
```
Comment (sethvargo) [inline website/source/docs/enterprise/backups/index.html.md]: Silly - two spaces after "and" here

File: website/source/docs/enterprise/backups/index.html.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+queries, sessions, and ACLs.
+
+[Consul Enterprise](https://www.hashicorp.com/consul.html) provides a [highly
+available service](https://www.consul.io/docs/commands/snapshot/agent.html) that
+integrates with the snapshot API to automatically manage taking snapshots,
+perform rotation and send backup files  offsite to Amazon S3.
```
Comment (sethvargo) [inline website/source/docs/enterprise/backups/index.html.md]: Two spaces after "files"

File: website/source/docs/enterprise/read-scale/index.html.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  Consul Enterprise supports increased read scalability without impacting write latency.
+---
+
+# Consul Enterprise Enhanced Read Scalability
+
+In [Consul Enterprise](https://www.hashicorp.com/consul.html), servers can be
+explicitly marked as non-voters. Non-voters will receive the replication stream
+but will not take part in quorum (required by the leader before log entries can
+be committed). Adding explicit non-voters will [scale
+reads](https://www.consul.io/docs/guides/autopilot.html#server-read-scaling)
```
Comment (sethvargo) [inline website/source/docs/enterprise/read-scale/index.html.md]: When linking to something internal, try to use relative URLs instead, so consider changing this to:

```
/docs/guides/autopilot.html#server-read-scaling
```

This is for two reasons:

1. In local dev, the links still work
1. Modern browsers will render that page faster because it doesn't need to lookup the DNS

Comment (sethvargo) [inline website/source/docs/enterprise/read-scale/index.html.md]: I saw a few places like this, but I'll only comment here.
